spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/TaskServiceUtils.java

```diff
@@ -398,4 +399,28 @@ static boolean isUseUserAccessToken(TaskConfigurationProperties taskConfiguratio
 
 		return taskConfigurationProperties.isUseUserAccessToken();
 	}
+
+	/**
+	 * Move and configure app arguments to the composed-task-app-arguments argument.  Also convert app args to base64 if necessary.
+	 * @param commandLineArgs The command line arguments to be converted
+	 * @return list of command line args where app args are now associated with the comosed-task-app arguments and all other remain unchanged.
+	 */
+	static List<String> convertCommandLineArgsToCTRFormat(List<String> commandLineArgs) {
+		List<String> composedTaskArguments = new ArrayList<>();
+		commandLineArgs.forEach(arg -> {
+			if (arg.startsWith("app.")) {
+				String[] split = arg.split("=", 2);
+				if (split.length == 2) {
+					composedTaskArguments.add("--composed-task-app-arguments." + Base64Utils.encode(split[0]) + "=" + split[1]);
```

Comment:
In this case, I think that
```java
Base64Utils.encode(split[0]) + "=" + split[1])
```
is the same as 
```java
Base64Utils.encode(arg)
```
Not actually suggesting a change - just pointing it out and thinking about it.

Reply:
This is correct because we only encode name of the child app and the index extension `.demo.0`.   The output of this would be to convert `app.demo.0=foo=bar` to `--composed-task-app-arguments.base64_YXBwLmRlbW8uMA=foo=bar`